src/if_statements.py

```diff
@@ -570,6 +603,216 @@ def try_make_if_condition(
+                    node.block.index > node.conditional_edge.block.index
+                    or context.options.compiler == context.options.CompilerEnum.GCC
+                ):
+                    if cond.right.value in cases:
+                        return None
+                    cases[cond.right.value] = node.fallthrough_edge
+                    node_queue.append(node.conditional_edge)
+                elif cond.op == "<" or (
+                    cond.op in (">=", "<=", ">")
+                    and context.options.compiler == context.options.CompilerEnum.GCC
+                ):
+                    # TODO: GCC (maybe?) can emit a series of `<`/`>=` comparisons that limit
+                    # to a specific case. Ex: `x < 9 && x >= 8` is only true for `x == 8`.
+                    # Detecting these would require tracking & checking all of the bounds.
+                    # For now, ignore those, and assume the compiler isn't emitting nonsense.
+                    # See PM: func_802403D4_97BA04
+                    node_queue.append(node.fallthrough_edge)
+                    node_queue.append(node.conditional_edge)
+                else:
+                    return None
+                irregular_comparison_count += 1
+                nodes_to_mark_emitted.add(node)
+                continue
+
+        elif isinstance(node, SwitchNode):
+            # The switch must use the same control expression
+            if block_info.switch_control is None:
+                return None
+            if early_unwrap_ints(block_info.switch_control.control_expr) != uw_var_expr:
+                return None
+            # Add the cases from the inner switch to our dict of cases
+            for i, case in enumerate(
+                node.cases, start=block_info.switch_control.offset
+            ):
+                if i in cases:
+                    return None
+                cases[i] = case
+            nodes_to_mark_emitted.add(node)
+            irregular_comparison_count += 1
+            continue
+
+        # If we've gotten here, then the node is not a valid jump target for the switch,
+        # unless it could be the `default:`-labeled node.
+        if default_node is not None:
+            return None
+        if isinstance(node, ReturnNode) or (
+            start in node.dominators and end in node.postdominators
+        ):
+            default_node = node
+
+    # Need at least two irregular comparisons (to skip the regular ConditionalNode guard + SwitchNode pairs)
+    # Need to combine at least 3 nodes into 2 distinct cases, otherwise it could be a plain if/else with ||
+    if (
+        irregular_comparison_count < 2
+        or len(nodes_to_mark_emitted) < 3
+        or len(set(cases.values())) < 2
+    ):
+        return None
+
+    # If this new irregular switch uses all of the other switch nodes in the function,
+    # then we no longer need to add labelling comments with the switch_index
```

Comment:
(since the change to emit structured switches, are switch indices still actually useful?) Would be nice to clean this up so ordering is consecutive, but it seems like low priority.

Reply:
There are still a few situations where the case labels get mangled a bit, primarily if the original source uses `goto`.